Signal/src/phone/PhoneManager.m

```diff
@@ -139,7 +139,7 @@ - (void)incomingCallWithSession:(ResponderSessionDescriptor *)session {
 
 - (BOOL)hasOngoingCall
 {
-    return self.curCallController != nil;
+    return [currentCallControllerObservable.currentValue callState].futureTermination.isIncomplete;
```

Comment:
You were right. The test I was using remained true after a red phone call is hung up.